pkg/controller/v1alpha2/applicationconfiguration/render_test.go

```diff
@@ -1066,8 +1066,168 @@ func TestRenderTraitName(t *testing.T) {
 	assert.NoError(t, err)
 	data := unstructured.Unstructured{Object: mapResult}
 
-	traitName := getTraitName(ac, componentName, &ac.Spec.Components[0].Traits[0], &data)
-	assert.Equal(t, traitName, "component-trait-11111111")
+	traitName := getTraitName(ac, componentName, &ac.Spec.Components[0].Traits[0], &data, &v1alpha2.TraitDefinition{})
+	assert.Equal(t, traitName, "component3-trait-11111111")
+}
+
+func TestRenderTraitNameWithoutReferenceName(t *testing.T) {
+	var scheme = runtime.NewScheme()
+	assert.NoError(t, clientgoscheme.AddToScheme(scheme))
+	assert.NoError(t, core.AddToScheme(scheme))
+	namespace := "ns"
+	acName := "coolappconfig4"
+	acUID := types.UID("definitely-a-uuid")
+	componentName := "component4"
+
+	mts := v1alpha2.ManualScalerTrait{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+		},
+		Spec: v1alpha2.ManualScalerTraitSpec{
+			ReplicaCount: 3,
+		},
+	}
+
+	gvks, _, _ := scheme.ObjectKinds(&mts)
```

Comment:
```
gvk:=gvks[0]
```

and use `gvk` instead of `gvks[0]` below.

We don't like hard code so many places even in test code.

Reply:
Okay, corrected all of them throughout the _test.go.